### ISSUE TITLE
What is the meaning of the `trim` parameter in the `assemble` command. Trim value 1 and 9 produced identical consensus sequences.

### ISSUE THREAD

**Issue by TatyanaVl** [gear-genomics/tracy#43]
What are the units of the  trimming stringency  in the `assemble` command?
`-t [ --trim ] arg (=4)              trimming stringency [1:9]`
What does 1-9 mean and may you please provide short description or refer us to the description of the trimming algorithm?

I have done assembly using different  trimming stringency and got the following results:
```
	trim	length	f_leadingGaps	f_mid_gaps	m_f_gap_indexes	f_trailingGaps	r_leadingGaps	rev_mid_gaps	m_r_gap_indexes	r_trailingGaps	total_gaps	sequence
0	1	348	39	2	[4, 9]	0	0	0	[]	42	41	"TTTGATCGTGGCTCAGGACGAACGCTGGCGGCGTGCCTAATACATGCAAGTAGAACGCTGAGAACTGGTGCTTGCACCGGTTCAAGGAGTTGCGAACGGGTGAGTAACGCGTAGGTAACCTACCTCATAGCGGGGGATAACTATTGGAAACGATAGCTAATACCGCATAAGAGAGACTAACGCATGTTAGTAATTTAAAAGGGGCAATTGCTCCACTATGAGATGGACCTGCGTTGTATTAGCTAGTTGGTGAGGTAAAGGCTCACCAAGGCGACGATACATAGCCGACCTGAGAGGGTGATCGCCCACACTGGGACTGAGACACGGCCCAGACTCCTACGGGCGGC"
1	2	315	133	0	[]	0	0	0	[]	120	133	"CGAACGCTGGCGGCGTGCCTAATACATGCAAGTAGAACGCTGAGAACTGGTGCTTGCACCGGTTCAAGGAGTTGCGAACGGGTGAGTAACGCGTAGGTAACCTACCTCATAGCGGGGGATAACTATTGGAAACGATAGCTAATACCGCATAAGAGAGACTAACGCATGTTAGTAATTTAAAAGGGGCAATTGCTCCACTATGAGATGGACCTGCGTTGTATTAGCTAGTTGGTGAGGTAAAGGCTCACCAAGGCGACGATACATAGCCGACCTGAGAGGGTGATCGGCCACACTGGGACTGAGACACGGCCCAG"
2	3	320	57	0	[]	0	0	0	[]	52	57	"GGACGAACGCTGGCGGCGTGCCTAATACATGCAAGTAGAACGCTGAGAACTGGTGCTTGCACCGGTTCAAGGAGTTGCGAACGGGTGAGTAACGCGTAGGTAACCTACCTCATAGCGGGGGATAACTATTGGAAACGATAGCTAATACCGCATAAGAGAGACTAACGCATGTTAGTAATTTAAAAGGGGCAATTGCTCCACTATGAGATGGACCTGCGTTGTATTAGCTAGTTGGTGAGGTAAAGGCTCACCAAGGCGACGATACATAGCCGACCTGAGAGGGTGATCGGCCACACTGGGACTGAGACACGGCCCAGAC"
3	4	327	39	0	[]	0	0	0	[]	46	39	"CTCAGGACGAACGCTGGCGGCGTGCCTAATACATGCAAGTAGAACGCTGAGAACTGGTGCTTGCACCGGTTCAAGGAGTTGCGAACGGGTGAGTAACGCGTAGGTAACCTACCTCATAGCGGGGGATAACTATTGGAAACGATAGCTAATACCGCATAAGAGAGACTAACGCATGTTAGTAATTTAAAAGGGGCAATTGCTCCACTATGAGATGGACCTGCGTTGTATTAGCTAGTTGGTGAGGTAAAGGCTCACCAAGGCGACGATACATAGCCGACCTGAGAGGGTGATCGGCCACACTGGGACTGAGACACGGCCCAGACTCC"
4	5	336	40	0	[]	0	0	0	[]	49	40	"GTGGCTCAGGACGAACGCTGGCGGCGTGCCTAATACATGCAAGTAGAACGCTGAGAACTGGTGCTTGCACCGGTTCAAGGAGTTGCGAACGGGTGAGTAACGCGTAGGTAACCTACCTCATAGCGGGGGATAACTATTGGAAACGATAGCTAATACCGCATAAGAGAGACTAACGCATGTTAGTAATTTAAAAGGGGCAATTGCTCCACTATGAGATGGACCTGCGTTGTATTAGCTAGTTGGTGAGGTAAAGGCTCACCAAGGCGACGATACATAGCCGACCTGAGAGGGTGATCGGCCACACTGGGACTGAGACACGGCCCAGACTCCTACGG"
5	6	340	40	1	[3]	0	0	0	[]	49	41	"TCGTGGCTCAGGACGAACGCTGGCGGCGTGCCTAATACATGCAAGTAGAACGCTGAGAACTGGTGCTTGCACCGGTTCAAGGAGTTGCGAACGGGTGAGTAACGCGTAGGTAACCTACCTCATAGCGGGGGATAACTATTGGAAACGATAGCTAATACCGCATAAGAGAGACTAACGCATGTTAGTAATTTAAAAGGGGCAATTGCTCCACTATGAGATGGACCTGCGTTGTATTAGCTAGTTGGTGAGGTAAAGGCTCACCAAGGCGACGATACATAGCCGACCTGAGAGGGTGATCGGCCACACTGGGACTGAGACACGGCCCAGACTCCTACGGGC"
6	7	343	40	1	[5]	0	0	0	[]	47	41	"GATCGTGGCTCAGGACGAACGCTGGCGGCGTGCCTAATACATGCAAGTAGAACGCTGAGAACTGGTGCTTGCACCGGTTCAAGGAGTTGCGAACGGGTGAGTAACGCGTAGGTAACCTACCTCATAGCGGGGGATAACTATTGGAAACGATAGCTAATACCGCATAAGAGAGACTAACGCATGTTAGTAATTTAAAAGGGGCAATTGCTCCACTATGAGATGGACCTGCGTTGTATTAGCTAGTTGGTGAGGTAAAGGCTCACCAAGGCGACGATACATAGCCGACCTGAGAGGGTGATCGGCCACACTGGGACTGAGACACGGCCCAGACTCCTACGGGCG"
7	8	345	40	1	[6]	0	0	0	[]	45	41	"TGATCGTGGCTCAGGACGAACGCTGGCGGCGTGCCTAATAAATGCAAGTAGAACGCTGAGAACTGGTGCTTGCACCGGTTCAAGGAGTTGCGAACGGGTGAGTAACGCGTAGGTAACCTACCTCATAGCGGGGGATAACTATTGGAAACGATAGCTAATACCGCATAAGAGAGACTAACGCATGTTAGTAATTTAAAAGGGGCAATTGCTCCACTATGAGATGGACCTGCGTTGTATTAGCTAGTTGGTGAGGTAAAGGCTCACCAAGGCGACGATACATAGCCGACCTGAGAGGGTGATCGGCCACACTGGGACTGAGACACGGCCCAGACTCCTACGGGCGG"
8	9	348	39	2	[4, 9]	0	0	0	[]	42	41	"TTTGATCGTGGCTCAGGACGAACGCTGGCGGCGTGCCTAATACATGCAAGTAGAACGCTGAGAACTGGTGCTTGCACCGGTTCAAGGAGTTGCGAACGGGTGAGTAACGCGTAGGTAACCTACCTCATAGCGGGGGATAACTATTGGAAACGATAGCTAATACCGCATAAGAGAGACTAACGCATGTTAGTAATTTAAAAGGGGCAATTGCTCCACTATGAGATGGACCTGCGTTGTATTAGCTAGTTGGTGAGGTAAAGGCTCACCAAGGCGACGATACATAGCCGACCTGAGAGGGTGATCGCCCACACTGGGACTGAGACACGGCCCAGACTCCTACGGGCGGC"
```

Trim `1 `and trim `9` produced identical consensus sequences, which does not make sense.
Another interesting observation is that trim  value `2` has a very high values for the `leadingGaps` and `trailingGaps` both for F and R strands.

May you please provide short description of the trimming algorithm to understand whats going on. 
Thank you!

**Comment by tobiasrausch** [comment]
tracy can assemble dozens of traces and that makes it difficult to specify a set left and right trimming length for each input trace. Instead, tracy uses '-t' which specifies a trimming stringency, with 1 being the most aggressive trimming and 9 being almost no trimming. In simple terms, the algorithm works like a match extension (or x-drop extension), the best quality trace segment is extended as long as the quality doesn't deteriorate too much. Looking at your results, the length of the sequence indeed suggests that '-t 9' gave the longest sequence and then it was trimmed more and more aggressively with '-t 2' giving the shortest sequence. '-t 1' looks indeed odd. I suspect, it probably was too stringent but this should yield an error message. Did you maybe miss the error message? Otherwise this could be a bug and I am happy to look at your data if you can share it with me.

**Comment by TatyanaVl** [comment]
Thank you for your answer. 
- I doubled checked the results and yes the -t 1 generated the error that I overlooked.
- Why the scale is 1-9? What are the units? The reason why I am asking is we are planing to use tracy to generate consensus sequence and afterwards cutadapt for quality trim https://cutadapt.readthedocs.io/en/v1.8.2/guide.html#quality-trimming, which has a wider range of quality trimming. However, that might not be necessary if  quality trim done by the cutadapt is redundant to quality trim done by tracy.
Thank you!!!
